### PR TITLE
v5.16.0

### PR DESCRIPTION
**Pull request by cherniavskii**
- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
      To do so, use `yarn release:changelog` The options are the following:

  ```bash
  yarn release:changelog
     --githubToken   YOUR_GITHUB_TOKEN (needs "public_repo" permission)
     --lastRelease   The release to compare against (default: the last one)
     --release       The branch to release (default: master)
  ```

  You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.

  In case of a problem, another method to generate the changelog is available at the end of this page.

- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version`.
- [x] Fix manually the package version in `x-date-pickers/package.json` and `x-date-pickers-pro/package.json`.
- [x] Open PR with changes and wait for review and green CI.
- [ ] Merge PR once CI is green, and it has been approved.

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Create a new tag named with the release you just did `git tag -a v4.0.0-alpha.30 -m "Version 4.0.0-alpha.30" && git push upstream --tag`

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```sh
git push upstream master:docs-v5 -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v5)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v5` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,61 @@
...
+- [docs] Fix typo in `migration from lab` (#5277) @chuckwired
+- [docs] Use `dayjs` instead of `date-fns` in doc examples (#5481) @flaviendelangle
+
+### Core
+
+- [core] Clarify the scope of this license key (#5824) @oliviertassinari
+- [core] Fix Typescript error on field hooks (#5892) @flaviendelangle
+- [core] Memoize `columns` in `useDemoData` hook (#5848) @cherniavskii
+- [core] Remove Firefox from the BrowserStack list (#5874) @DanailH
+- [core] Small changes to the release script (#5840) @m4theushw
```
Comment (LukasTy) [inline CHANGELOG.md:59]: Nitpick: what do you think about removing these from the changelog?
These are internal changes in no way related to what the release delivers... 🤔

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,61 @@
...
+
+### `@mui/x-date-pickers@v5.0.0-beta.7` / `@mui/x-date-pickers-pro@v5.0.0-beta.7`
+
+#### Changes
+
+- [DatePicker] Fix to pass down `className` prop provided on DatePicker to `renderInput` (#5471) @CruseCtrl
+- [DatePicker] Improve `a11y` support (#5809) @LukasTy
+- [pickers] Add `PaperContent` component slot (#5801) @LukasTy
+- [pickers] Add a breaking change section in the migration guide (#5805) @alexfauquette
+- [pickers] Add key in the `localeText` (#5143) @alexfauquette
```
Comment (flaviendelangle) [inline CHANGELOG.md:41]:
```suggestion
- [pickers] Add new translations to `localeText` (#5143) @alexfauquette
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,61 @@
...
+- [l10n] Add Polish (pl-PL) locale (#5833) @drastus
+
+### Docs
+
+- [docs] Fix typo in `migration from lab` (#5277) @chuckwired
+- [docs] Use `dayjs` instead of `date-fns` in doc examples (#5481) @flaviendelangle
+
+### Core
+
+- [core] Clarify the scope of this license key (#5824) @oliviertassinari
```
Comment (flaviendelangle) [inline CHANGELOG.md:55]:
```suggestion
- [core] Clarify the scope of the license key used for tests and documentation (#5824) @oliviertassinari
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,61 @@
...
+
+_Aug 25, 2022_
+
+We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:
+
+- 🎁 Introduce column grouping for data grid (#5133) @alexfauquette
+
+  You can now group columns using the `columnGroupingModel` prop. This lets you to display more structured data.
+
+  ![image](https://user-images.githubusercontent.com/45398769/186178366-4fba66b2-bf90-4c7a-9d83-940a7fc78704.png)
```
Comment (m4theushw) [inline CHANGELOG.md:16]: The screenshot is blurry at full width.

```suggestion
  <img src="https://user-images.githubusercontent.com/45398769/186178366-4fba66b2-bf90-4c7a-9d83-940a7fc78704.png" width="800" />
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,61 @@
...
+
+  To enable this feature, add `experimentalFeatures={{ columnGrouping: true }}`.
+  The grouping header can be fully customized.
+  See the [documentation](https://mui.com/x/react-data-grid/column-groups/) to explore everything it has to offer.
+
+- 🧪 Add new unstable [`DateField`](https://mui.com/x/react-date-pickers/date-field/) and [`DateRangeField`](https://mui.com/x/react-date-pickers/date-range-field/) components for date pickers (#5504) @flaviendelangle
```
Comment (cherniavskii) [inline CHANGELOG.md:22]: @flaviendelangle is it a good time to include these components in the highlights? Or is it too early?

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,61 @@
...
+### `@mui/x-date-pickers@v5.0.0-beta.7` / `@mui/x-date-pickers-pro@v5.0.0-beta.7`
+
+#### Changes
+
+- [DatePicker] Fix to pass down `className` prop provided on DatePicker to `renderInput` (#5471) @CruseCtrl
+- [DatePicker] Improve `a11y` support (#5809) @LukasTy
+- [pickers] Add `PaperContent` component slot (#5801) @LukasTy
+- [pickers] Add a breaking change section in the migration guide (#5805) @alexfauquette
+- [pickers] Add key in the `localeText` (#5143) @alexfauquette
+- [pickers] Document components slots like for the data grid (#4657) @flaviendelangle
```
Comment (m4theushw) [inline CHANGELOG.md:42]:
```suggestion
- [pickers] Document components slots (#4657) @flaviendelangle
```

File: CHANGELOG.md (hunk 5 above)
Comment (flaviendelangle) [inline CHANGELOG.md:22]:
```suggestion
- 🧪 Add new unstable [`DateField`](https://mui.com/x/react-date-pickers/date-field/) and [`DateRangeField`](https://mui.com/x/react-date-pickers/date-range-field/) components for date pickers (#5504) @flaviendelangle

  ⚠️ Those components are very early stage and should not be used in production. The documentation will be published when they will be mature enough to be experimented by the community.
```

File: CHANGELOG.md (hunk 3 above)
Comment (m4theushw) [inline CHANGELOG.md:55]:
```suggestion
- [core] Clarify the scope of the license key (#5824) @oliviertassinari
```

**Review by m4theushw — COMMENTED** [review]
Don't forget to bump the version.

File: CHANGELOG.md (hunk 5 above)
Comment (flaviendelangle) [reply CHANGELOG.md:22]: On one side they are to early stage to be used in a real application so we should not advertise about them

On the other side, maybe it's better to be explicit and say "Hey they exist, but be careful" and to start to gather feedback from very early testers (but if we have too many they will be tons of feedbacks like "Mobile is broken" which we know)

Comment (alexfauquette) [reply CHANGELOG.md:22]: For now, we will get too many issues if users try to use it. Better to highlight this when we will have fix most of the issues we are already aware about.

Moreover we have no documentation, so they will struggle to use the component before realizing all the limitations

File: CHANGELOG.md (hunk 1 above)
Comment (cherniavskii) [reply CHANGELOG.md:59]: Historically we include everything in the release notes.
I have no strong preference on this, but sometimes it helps to find some clues what caused a regression

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,61 @@
...
+
+  You can now group columns using the `columnGroupingModel` prop. This lets you to display more structured data.
+
+  <img src="https://user-images.githubusercontent.com/45398769/186178366-4fba66b2-bf90-4c7a-9d83-940a7fc78704.png" width="800" />
+
+  To enable this feature, add `experimentalFeatures={{ columnGrouping: true }}`.
+  The grouping header can be fully customized.
+  See the [documentation](https://mui.com/x/react-data-grid/column-groups/) to explore everything it has to offer.
+
+- 🐞 Bugfixes
```
Comment (alexfauquette) [inline CHANGELOG.md:22]:
```suggestion
- 🐞 Bugfixes

- 🌏 New locales thanks to @tesseractjh and @drastus
```

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md (hunk 7 above)
Comment (LukasTy) [inline CHANGELOG.md:22]:
```suggestion
- 🐞 Bugfixes
- 📚 Documentation improvements
- 🌏 New locales thanks to @tesseractjh and @drastus
```
🙈

**Review by LukasTy — APPROVED** [review]

**Review by DanailH — APPROVED** [review]

**Review by flaviendelangle — APPROVED** [review]

**Pull request merged**